src/librustc_hir/def.rs

```diff
@@ -65,6 +65,10 @@ pub enum DefKind {
     AssocOpaqueTy,
     TyParam,
 
+    ExternCrate,
+    Use,
+    ForeignMod,
```

Comment:
Please, list all the new variants as not namespaced and return `false` for them (or panic) in `matches_ns`.
It should never matter what namespace they return and would be good to ensure that it's not used.